### ISSUE TITLE
The loss in trades.py

### ISSUE THREAD

**Issue by wysroy** [yaodongyu/TRADES#5]
According to the paper eq. (5), the loss L should be the same, saying cross entropy loss in the paper. While in trades.py, there are two kinds of loss used. For max, torch.nn.KLDivLoss is adopted, while for min, cross entropy used for f(x),y and torch.nn.KLDivLoss used for f(x),f(x'). So why use two different loss here? using both cross entropy loss at the both place is ok? performance?

**Comment by yaodongyu** [comment]
Thanks for proposing this.

Q: 'For max, torch.nn.KLDivLoss is adopted' 
A: For the maximization problem, using torch.nn.KLDivLoss and cross entropy is equivalent because they differ by a constant additive term.

Q: 'while for min, cross entropy used for f(x),y and torch.nn.KLDivLoss used for f(x),f(x'). So why use two different loss here? using both cross entropy loss at both place is ok? performance?'
A: For the minimization problem, it is okay to use both cross entropy losses at both places, and the performance of using both cross entropy is as good as using the combination of KL divergence loss and cross entropy loss. 
By the way, the cross-entropy loss for f(x) can be decomposed as an KL divergence loss term plus a constant entropy term, i.e., CE(f(X), Y) = KL(f(X)||Y) + H(Y). Therefore, the two different losses can be regarded as two KL divergence losses.

**Comment by jh-jeong** [comment]
> By the way, the cross-entropy loss for f(x) can be decomposed as an KL divergence loss term plus a constant entropy term, i.e., CE(f(X), Y) = KL(f(X)||Y) + H(Y). Therefore, the two different losses can be regarded as two KL divergence losses.

@yaodongyu  

But.. isn't that the entropy H(Y) is not a constant when used to minimize CE(f(x), f(x'))?
